File: two_phase_flow_complex_domain/post_process/test_2.py
import numpy as np
import matplotlib.pyplot as plt
from da1 import da1
from da2_snake import da2_snake

##
# read multiple data files
lx = 1.0
ly = 1.0

data_path = r'D:\numerical_simulation\share\mcl_complex\tt\\'

m = 128
n = 128

# Use da2_snake to process and plot the data sequence
x_lst, y_lst, mass_lst, Area_lst, t_lst, u_c_lst = da2_snake(
    0, 1, 3, m, n, data_path, '', lx, ly, False
)


##
# read single data files
lx = 1.0
ly = 1.0

data_path = r'D:\numerical_simulation\share\mcl_complex\tt\\'

m = 128
n = 128
r, z, u, v, p, f, f2, t = da1(data_path + 'data003', n + 1, m + 1) # , m + 1, n + 1)

# Plot
plt.figure()
plt.title(f't = {t}')
plt.contour(r, z, f, levels=[0.5], colors='m')
plt.contour(r, z, f2, levels=[0.5], colors='m')
plt.axis('equal')
plt.title(f't = {t}')

ip = 5
scale_factor = 0.1  # Adjust arrow size

# Downsample for quiver plot
r_q = r[::ip, ::ip]
z_q = z[::ip, ::ip]
u_q = u[::ip, ::ip] * scale_factor
v_q = v[::ip, ::ip] * scale_factor

plt.quiver(r_q, z_q, u_q, v_q, color='r', angles='xy', scale_units='xy', scale=1)
#plt.quiver(r, z, u, v)
plt.show()

File: two_phase_flow_complex_domain/post_process/test_2.m


%%
% read multiple data files
clc; clear; close all;

lx = 1.0; ly = 1.0;

data_path = 'D:\numerical_simulation\share\mcl_complex\tt\';

m=128;n=128;
[x_lst, y_lst, mass_lst, Area_lst, t_lst, u_c_lst] = da2_snake(0,1,12, m,n, data_path,'', lx,ly,false);



%%
% read single data file
clc; clear; close all;

lx = 1.0; ly = 1.0;
data_path = 'D:\numerical_simulation\share\mcl_complex\tt\';

m=128;n=128;

[r,z,u,v,p,f,f2,t]=da1([data_path 'data003'],m+1,n+1);
% plot
figure; hold on;
title(['t=' num2str(t)]);
contour(r,z,f,[0.5 0.5], 'm'); hold on;
contour(r,z,f2,[0.5 0.5], 'm');
axis equal;
title(['t = ' num2str(t)])

ip = 5;
scale_factor = 0.1; % 缩放系数，调整箭头的大小
quiver(r(1:ip:end,1:ip:end),z(1:ip:end,1:ip:end),...
    u(1:ip:end,1:ip:end)*scale_factor,v(1:ip:end,1:ip:end)*scale_factor,'r','AutoScale', 'off');
%quiver(r,z,u,v);